### PR TITLE
#1 - create db tables

### PR DESCRIPTION
**Pull request by MaciejSocha**
Created db tables from google doc. First version.

image

### CODE REVIEW

File: src/main/java/razdwatrzy/zzpj/model/Bonuses.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
+package razdwatrzy.zzpj.model;
+
+import javax.persistence.*;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Size;
+
+@Entity
+@Table(name = "bonuses")
+public class Bonuses {
```
Comment (stanislaw-zakrzewski) [inline src/main/java/razdwatrzy/zzpj/model/Bonuses.java]: Bonuses -> Bonus

File: src/main/java/razdwatrzy/zzpj/model/Campaign.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    @Size(max = 16)
+    @Column(unique = true)
+    private String title;
+
+    @NotNull
+    @Size(max = 32)
+    @Column
+    private String shortDescription;
+
+    @NotNull
```
Comment (stanislaw-zakrzewski) [inline src/main/java/razdwatrzy/zzpj/model/Campaign.java:42]: @AconLight I think it shouldn't be so short.

File: src/main/java/razdwatrzy/zzpj/model/Campaign.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    @Column
+    private Boolean isFinished;
+
+    @NotNull
+    @Size(max = 16)
+    @Column(unique = true)
+    private String title;
+
+    @NotNull
+    @Size(max = 32)
```
Comment (stanislaw-zakrzewski) [inline src/main/java/razdwatrzy/zzpj/model/Campaign.java:38]: @AconLight I think it shouldn't be so short.

File: src/main/java/razdwatrzy/zzpj/model/Campaign.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    @NotNull
+    @Column
+    private int pointsToWin;
+
+    @NotNull
+    @Column
+    private Boolean isFinished;
+
+    @NotNull
+    @Size(max = 16)
```
Comment (stanislaw-zakrzewski) [inline src/main/java/razdwatrzy/zzpj/model/Campaign.java:33]: @AconLight I think it shouldn't be so short.

File: src/main/java/razdwatrzy/zzpj/model/User.java [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+@Entity
+@Table(name = "users")
+public class User {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private long id;
+
+    @NotNull
+    @Size(max = 256)
```
Comment (stanislaw-zakrzewski) [inline src/main/java/razdwatrzy/zzpj/model/User.java]: Maybe this should be shorter ~32

File: src/main/java/razdwatrzy/zzpj/model/UserCampaign.java [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    @JoinColumn(name = "campaign_id", nullable = false)
+    private Campaign campaign;
+
+    @OneToOne(fetch = FetchType.LAZY, optional = false)
+    @JoinColumn(name = "user_id", nullable = false)
+    private User user;
+
+    @NotNull
+    @Column
+    private long parentID;
```
Comment (stanislaw-zakrzewski) [inline src/main/java/razdwatrzy/zzpj/model/UserCampaign.java:23]: @AconLight What is it for?

File: src/main/java/razdwatrzy/zzpj/model/UserCredentials.java [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+    private User user;
+
+    @NotNull
+    @Email
+    @Column(unique = true)
+    private String email;
+
+    @NotNull
+    @Column
+    private String password;
```
Comment (stanislaw-zakrzewski) [inline src/main/java/razdwatrzy/zzpj/model/UserCredentials.java:26]: Is there a 'password" type in database, it'll be more secure.

**Review by stanislaw-zakrzewski — CHANGES_REQUESTED** [review]

File: src/main/java/razdwatrzy/zzpj/model/Bonuses.java (hunk 1 above)
Comment (MaciejSocha) [reply src/main/java/razdwatrzy/zzpj/model/Bonuses.java]: Repaierd

File: src/main/java/razdwatrzy/zzpj/model/User.java (hunk 5 above)
Comment (MaciejSocha) [reply src/main/java/razdwatrzy/zzpj/model/User.java]: Yes ;)

File: src/main/java/razdwatrzy/zzpj/model/UserCredentials.java (hunk 7 above)
Comment (MaciejSocha) [reply src/main/java/razdwatrzy/zzpj/model/UserCredentials.java:26]: I did not find anything like that

**Pull request merged**